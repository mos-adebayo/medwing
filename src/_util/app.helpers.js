import axios from "axios";
import moment from "moment/moment";
import {appConstants} from "../_constant";

export const appHelpers = {
    mapConfig : () => {
      return {
          source: appConstants.TOM_TOM_MAP_SOURCE,
          key: appConstants.TOM_TOM_KEY,
          basePath: '/sdk',
          zoom: appConstants.MAP_ZOOM_LEVEL
      }
    },
    mapSearchBoxConfig : () => {
      return {
          location: false,
          language: 'pl-PL',
          view: 'IN',
          position: appConstants.TOM_TOM_SEARCH_BOX_POSITION,
          serviceOptions: {unwrapBbox: true}
      }
    },
    formatDate: (d, format) => {
        return moment(d).format(format || "MMM DD, YYYY");
    },
    interpretError : error => {
        let errorMessage;
        if (error.response) {
            // The request was made and the server responded with a status code
            // that falls out of the range of 2xx
            if(error.response.status === 500){
                errorMessage = "Oops, something went wrong";
            }else if(error.response.status === 404){
                errorMessage = "Resource not found";
            }else{
                errorMessage = "Oops, unable to process request"
            }
        } else if (error.request) {
            // The request was made but no response was received
            // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
            // http.ClientRequest in node.js
            errorMessage = 'Connection refused';
        } else {
            // Something happened in setting up the request that triggered an Error
            errorMessage = error.message;
        }
        return errorMessage;
    },

    getRequest: function (url) {
        let reqHeader = { "Content-Type": "application/json" };
        let config = { headers: reqHeader };
        return axios
            .get(url, config)
            .then(function (res) {
                return {status: appConstants.REQUEST_SUCCESS, data: res.data}
            })
            .catch( (error) =>  {
                const errorMessage = this.interpretError(error);
                return { status: appConstants.REQUEST_ERROR, error: errorMessage}
            });
    },
    postRequest: function (url, payload) {
        let reqHeader = { "Content-Type": "application/json" };

        let config = { headers: reqHeader };

        return axios.post(url, payload, config)
            .then(res => {
                return {status: appConstants.REQUEST_SUCCESS, data: res.data}
            }).catch((error) => {
                const errorMessage = this.interpretError(error);
                return {status: appConstants.REQUEST_ERROR, error: errorMessage}
            });
    },
    putRequest: function (url, payload) {
        let reqHeader = { "Content-Type": "application/json" };

        let config = { headers: reqHeader };

        return axios.put(url, payload, config)
            .then(res => {
                return {status: appConstants.REQUEST_SUCCESS, data: res.data}
            }).catch((error) => {
                const errorMessage = this.interpretError(error);
                return {status: appConstants.REQUEST_ERROR, error: errorMessage}
            });
    },
    deleteRequest: function (url) {
        let reqHeader = { "Content-Type": "application/json" };

        let config = { headers: reqHeader };

        return axios.delete(url, config)
            .then(res => {
                return {status: appConstants.REQUEST_SUCCESS, data: res.data}
            }).catch((error) => {
                const errorMessage = this.interpretError(error);
                return {status: appConstants.REQUEST_ERROR, error: errorMessage}
            });
    },
  interpretErrorResponse(error) {
    let errorMessage = "";
    if(error.response){
        errorMessage = error.response.data
            ? error.response.data
            : "Unable to handle request";
    }else if(error.request){
        errorMessage = "Currently, unable to handle request!"
    }else{
        errorMessage = "Something went wrong!"
    }
    if (typeof errorMessage === "string") {
      return errorMessage;
    }
    else if (Array.isArray(errorMessage)) {
      return errorMessage.join(',');
    } else {
      return "Something went wrong!";
    }
  }
};
