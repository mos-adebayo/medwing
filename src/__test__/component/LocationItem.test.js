import React from "react";
import { configure, shallow } from "enzyme";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import LocationItem from "../../components/Location/locations.items";

//Enzyme Configurations
import Adapter from 'enzyme-adapter-react-16';
import {appHelpers} from "../../_util";
configure({ adapter: new Adapter() });


const location = {
    "createdAt": 1545500440790,
    "updatedAt": 1545638365904,
    "id": "5c1e77186455a4178b6d771d",
    "address": "Ahmadu Bello Way, Lagos, Lagos",
    "latitude": 6.440976,
    "longitude": 3.402617,
    "country": "Nigeria",
    "countryCode": "NG"
};

function setup() {
    const props = {
        location: location,
        locateMarker: jest.fn(),
        removeMarker: jest.fn()
    };

    const enzymeWrapper = shallow(
        <LocationItem {...props} />
    );

    return {
        props,
        enzymeWrapper
    }
}

describe('Location Item component', () => {
    describe('Location Item', () => {
        test('component snapshot', () => {
            const component = renderer.create(
                <LocationItem location={location} locateMarker={jest.fn()} removeMarker={jest.fn()}/>,
            );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });

        test('component receives props and then renders successfully', () => {
            const { enzymeWrapper } = setup();
            //confirm location latitude and longitude is rendered
            expect(enzymeWrapper.contains((
                <p className={'truncate'}>{location.address}</p>
            ))).toBeTruthy();
            //confirm date is rendered
            expect(enzymeWrapper.contains((
                <p className={'grey-text'}>
                    <small>{appHelpers.formatDate(location.createdAt)}</small>
                </p>
            ))).toBeTruthy();
        });

        test('should call remove marker callback when the remove icon is clicked ', () => {
            const { enzymeWrapper, props } = setup();
            const target = enzymeWrapper.find('.remove-icon');
            target.simulate('click');
            expect(props.removeMarker.mock.calls.length).toBe(1)
        });

        test('should call locate callback when the locate icon is triggered ', () => {
            const { enzymeWrapper, props } = setup();
            const target = enzymeWrapper.find('.locate-icon');
            target.simulate('click');
            expect(props.locateMarker.mock.calls.length).toBe(1)
        });
    })
})
