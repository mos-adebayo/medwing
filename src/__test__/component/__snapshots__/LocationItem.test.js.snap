// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Item component Location Item component snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <p
      className="grey-text"
    >
      <small>
        Address
      </small>
    </p>
    <p
      className="truncate"
    >
      Ahmadu Bello Way, Lagos, Lagos
    </p>
    <p
      className="grey-text"
    >
      <small>
        Latitude, Longitude
      </small>
    </p>
    <p>
      6.440976
      , 
      3.402617
    </p>
  </div>
  <div
    className="card-footer card-content grey lighten-4"
  >
    <div
      className="row"
    >
      <div
        className="left"
      >
        <a
          className="locate-icon"
          href="/"
          onClick={[Function]}
        >
          <i
            className="tiny material-icons"
          >
            search
          </i>
        </a>
        Â 
        <a
          className="text-red remove-icon"
          href="/"
          onClick={[Function]}
        >
          <i
            className="tiny material-icons red-text"
          >
            delete
          </i>
        </a>
      </div>
      <div
        className="right"
      >
        <p
          className="grey-text"
        >
          <small>
            Dec 22, 2018
          </small>
        </p>
      </div>
    </div>
  </div>
</div>
`;
